watch.go

```diff
@@ -127,3 +127,25 @@ func (w WatchSet) watchMany(ctx context.Context) error {
 		return ctx.Err()
 	}
 }
+
+// WatchChan returns a channel that is used to wait for either the watch set
+// to trigger or for the context to be cancelled.
+func (w WatchSet) WatchChan(ctx context.Context) <-chan error {
+	// Create the outgoing channel
+	triggerCh := make(chan error)
+
+	// Create a goroutine to collect the errors from WatchCtx
+	go func() {
+		for {
```

Comment:
All the other `Watch` methods fire exactly once. I would mimic this behavior and not use a loop here. Then you can change the whole implementation to `triggerCh <- w.WatchCtx(ctx)`